inference-engine/src/inference_engine/ie_core.cpp

```diff
@@ -700,6 +718,12 @@ class Core::Impl : public ICore {
 
                 // configuring
                 {
+                    if (DeviceSupportsCacheDir(plugin)) {
+                        auto cacheConfig = coreConfig.getCacheConfig();
+                        if (cacheConfig._cacheManager) {
+                            desc.defaultConfig[CONFIG_KEY(CACHE_DIR)] = cacheConfig._cacheDir;
```

Comment:
I see you updated `SetConfigForPlugins`, should not we just update this code as well inside `SetConfigForPlugins`:
```
desc.second.defaultConfig[conf.first] = conf.second;
```
But keep `GetCPPPluginByName` as is?

Reply:
Sorry, didn't get your comment. Per my understanding, `desc.second.defaultConfig` is for not-yet-loaded plugins, which are registered, but not loaded yet.

And when plugin is loaded later (using `GetCPPPluginByName`) - this 'deferred config' is applied.
So, for we should store 'CACHE_DIR' in deferred config (because we can't check if plugin supports CACHE_DIR when it is not loaded), but later, when plugin is loaded we shall check if 'CACHE_DIR' config is supported and then apply it.

Reply:
oh, I see.